backend/src/space.rs

```diff
@@ -0,0 +1,17 @@
+
+struct Space {
+    x: u64,
+    y: u64,
+    contains: SpaceKind,
+} 
+
+enum SpaceKind {
+    Kube(Kube),
+    Player(Player),
+    EmptySpace(EmptySpace),
```

Comment:
I don't think this is necessary? The coordinates are already known from the `Space` struct, so `EmptySpace` can just be an enum entry without holding any extra data, so I'd change this line to just be `EmptySpace,`.